client/extensions/woocommerce/app/promotions/fields/currency-field.js

```diff
@@ -0,0 +1,61 @@
+/**
+ * External dependencies
+ */
+import React from 'react';
+import PropTypes from 'prop-types';
+
+/**
+ * Internal dependencies
+ */
+import PriceInput from 'woocommerce/components/price-input';
+import FormField from './form-field';
+
+const CurrencyField = ( {
+	fieldName,
+	labelText,
+	explanationText,
+	placeholderText,
+	isRequired,
+	value,
+	edit,
+	currency,
+} ) => {
+	const renderedValue = ( 'undefined' !== typeof value ? value : '' );
+
+	const onChange = ( e ) => {
+		const newValue = e.target.value;
+		// TODO: Round to correct decimal number count (e.g. 2.505 -> 2.51).
```

Comment:
You should be able to use `getCurrencyFormatDecimal( value, storeCurrency )` from `woocommerce/lib/currency`, and it will round to the right precision for that currency.

Reply:
I remembered something about this work being done, thanks for the reminder!